src/cmd/health.js

```diff
@@ -129,24 +134,61 @@ class NSFSHealth {
             name: NSFS_SERVICE,
             service_status: service_status,
             pid: pid,
+            error_type: health_errors_tyes.PERSISTENT,
           },
           {
             name: RSYSLOG_SERVICE,
             service_status: rsyslog.service_status,
             pid: rsyslog.pid,
+            error_type: health_errors_tyes.PERSISTENT,
           }],
           endpoint: {
-            endpoint_state
+            endpoint_state,
+            error_type: health_errors_tyes.TEMPORARY,
           },
-          invalid_accounts: account_details.invalid_storages,
-          valid_accounts: account_details.valid_storages,
-          invalid_buckets: bucket_details.invalid_storages,
-          valid_buckets: bucket_details.valid_storages,
+          accounts_status: {
+            invalid_accounts: account_details.invalid_storages,
+            valid_accounts: account_details.valid_storages,
+            error_type: health_errors_tyes.PERSISTENT,
+          },
+          buckets_status: {
+            invalid_buckets: bucket_details.invalid_storages,
+            valid_buckets: bucket_details.valid_storages,
+            error_type: health_errors_tyes.PERSISTENT,
+          }
       }
     };
+    if (!this.all_account_details) {
+      delete health.checks.accounts_status;
+    }
+    if (!this.all_bucket_details) {
+      delete health.checks.buckets_status;
+    }
     return health;
   }
 
+  async get_endpoint_response() {
+    let endpoint_state;
+    try {
+      await P.retry({
+        attempts: config.NSFS_HEALTH_ENDPOINT_RETRY_COUNT,
```

Comment:
are these new properties? I don't see they were added to config.js or if they exists already

Reply:
@romayalon missed this property after rebase :), Will add